src/core/connection.c

```diff
@@ -5618,9 +5620,7 @@ QuicConnRecvDatagrams(
                     Packet->BufferLength - (uint16_t)(Packet->AvailBuffer - Packet->Buffer);
             }
 
-            if (Connection->Crypto.CertValidationPending ||
-                Connection->Crypto.TicketValidationPending ||
-                !QuicConnRecvHeader(
```

Comment:
I am not 100% sure about this change, the original code seems to cause all incoming packets to be dropped while we have some pending validation, and thus we end up ignoring incoming CONNECTION_CLOSE. Was this intended?

Reply:
The condition was originally introduced by https://github.com/microsoft/msquic/pull/3186

Reply:
I'm not so sure either. I will follow up.

Reply:
The failing tests seem to give a clue that this change breaks something.

When we do async validation, we pretend that the cert is valid at the TLS layer and up the ReadKey on server side, so we start receiving and processing incoming 1RTT packets

```
Error: [03/01/2024 14:50:45] Handshake/WithHandshakeArgs5.CustomClientCertificateValidation/1 failed (in 2.1450049 sec):
Note: Google Test filter = Handshake/WithHandshakeArgs5.CustomClientCertificateValidation/1
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
Initializing for User Mode tests
[----------] 1 test from Handshake/WithHandshakeArgs5
[ RUN      ] Handshake/WithHandshakeArgs5.CustomClientCertificateValidation/1
D:\a\msquic\msquic\src\test\lib\HandshakeTest.cpp(916): error: Unexpected new Stream received

[  FAILED  ] Handshake/WithHandshakeArgs5.CustomClientCertificateValidation/1, where GetParam() = Reject/Async (2084 ms)
[----------] 1 test from Handshake/WithHandshakeArgs5 (2084 ms total)
```

Reply:
Removing the check also caused asserts on Linux because MsQuic attempted to process CRYPTO frames while we have pending async validation.